js/plugins/iq.js

```diff
@@ -0,0 +1,352 @@
+          var currentPlayheadPositionMilli = currentPlayheadPosition * 1000;
+          this.ooyalaReporter.reportSeek(currentPlayheadPositionMilli, seekedPlayheadPositionMilli);
+          OO.log("IQ: Reported: reportSeek() with args: " + currentPlayheadPositionMilli + ", " + seekedPlayheadPositionMilli);
+        }
+        break;
+      //OO.EVENTS.PLAYED -> OO.Analytics.EVENTS.PLAYBACK_COMPLETED.
+      case OO.Analytics.EVENTS.PLAYBACK_COMPLETED:
+        this.ooyalaReporter.reportComplete();
+        OO.log("IQ: Reported: reportComplete()");
+        break;
+      //OO.EVENTS.REPLAY -> OO.Analytics.EVENTS.VIDEO_REPLAY_REQUESTED.
+      case OO.Analytics.EVENTS.VIDEO_REPLAY_REQUESTED:
+        this.ooyalaReporter.reportReplay();
+        OO.log("IQ: Reported: reportReplay()");
+        break;
+
+      case OO.Analytics.EVENTS.AD_REQUEST:
+      case OO.Analytics.EVENTS.AD_REQUEST_SUCCESS:
+      case OO.Analytics.EVENTS.AD_SDK_LOADED:
+      case OO.Analytics.EVENTS.AD_SDK_LOAD_FAILURE:
+      case OO.Analytics.EVENTS.AD_BREAK_STARTED:
+      case OO.Analytics.EVENTS.AD_BREAK_ENDED:
+      case OO.Analytics.EVENTS.AD_POD_STARTED:
+      case OO.Analytics.EVENTS.AD_POD_ENDED:
+      case OO.Analytics.EVENTS.AD_STARTED:
+      case OO.Analytics.EVENTS.AD_ENDED:
+      case OO.Analytics.EVENTS.AD_SKIPPED:
+      case OO.Analytics.EVENTS.AD_ERROR:
+      case OO.Analytics.EVENTS.AD_REQUEST_EMPTY:
+      case OO.Analytics.EVENTS.AD_REQUEST_ERROR:
+      case OO.Analytics.EVENTS.AD_PLAYBACK_ERROR:
+      case OO.Analytics.EVENTS.AD_IMPRESSION:
+      case OO.Analytics.EVENTS.AD_SDK_IMPRESSION:
+      case OO.Analytics.EVENTS.AD_COMPLETED:
+      case OO.Analytics.EVENTS.AD_CLICKTHROUGH_OPENED:
+      case OO.Analytics.EVENTS.SDK_AD_EVENT:
+        var eventMetadata = params[0];
+        if(!eventMetadata)
+        {
+          eventMetadata = {};
+        }
+
+        if (eventMetadata.adEventName)
+        {
+          eventMetadata.adEventName = eventName + ":" + eventMetadata.adEventName;
+        }
+        else
+        {
+          eventMetadata.adEventName = eventName;
+        }
+        this.ooyalaReporter.reportCustomEvent(eventName, eventMetadata);
+        OO.log("IQ: Reported: reportCustomEvent() for event: " + eventName + " with args:" + JSON.stringify(eventMetadata));
+        break;
+      default:
+        break;
+    }
+  };
+
+  /**
+   * [Required Function] Clean up this plugin so the garbage collector can clear it out.
+   * @public
+   * @method IqPlugin#destroy
+   */
+  this.destroy = function ()
+  {
+    _framework = null;
+  };
+
+  /**
+   * Called when the SDK fails to load.
+   * @private
+   * @method NielsenAnalyticsPlugin#sdkLoadError
+   */
+  var sdkLoadError = function()
+  {
+    //Destroy and unregister
+    if (_.isString(id))
+    {
+      framework.unregisterPlugin(id);
+    }
+    this.destroy();
+  };
+
+  var trySetupAnalytics = OO._.bind(function()
+  {
+    if (window.Ooyala)
```

Comment:
What does this check?

Reply:
@pilievOoyala any thoughts here? Not sure if this is related to that else if (!window.Ooyala) in the init function.

Reply:
I don't remember this at all.

Reply:
The only thing I can think of is people double loading the file.